src/history/base.rs

```diff
@@ -178,6 +178,45 @@ pub trait History: Send {
     fn sync(&mut self) -> std::io::Result<()>;
 }
 
+impl<T: History + ?Sized> History for Box<T> {
+    fn save(&mut self, h: HistoryItem) -> Result<HistoryItem> {
+        (**self).save(h)
+    }
+
+    fn load(&self, id: HistoryItemId) -> Result<HistoryItem> {
+        (**self).load(id)
+    }
+
+    // should this be changed?
+    fn count(&self, query: SearchQuery) -> Result<i64> {
+        (**self).count(query)
+    }
+
+    fn search(&self, query: SearchQuery) -> Result<Vec<HistoryItem>> {
+        (**self).search(query)
+    }
+
+    fn update(
+        &mut self,
+        id: HistoryItemId,
+        updater: &dyn Fn(HistoryItem) -> HistoryItem,
+    ) -> Result<()> {
+        (**self).update(id, updater)
+    }
+
+    fn clear(&mut self) -> Result<()> {
+        (**self).clear()
+    }
+
+    fn delete(&mut self, h: HistoryItemId) -> Result<()> {
+        (**self).delete(h)
+    }
+
+    fn sync(&mut self) -> std::io::Result<()> {
+        (**self).sync()
+    }
+}
+
```

Comment:
I don't see the reason for reimplementing `History` for `Box<impl History>` as the trait is wholy defined for working with references to the `History` which is trivially obtainable from a boxed value.